h/static/scripts/hooks/use-group-annotations.tsx

```diff
@@ -44,6 +48,12 @@ export type GroupAnnotationsResult = {
    */
   removedAnnotations: Set<string>;
 
+  /**
+   * Number of annotations in the `annotations` list which are not in the
+   * `removedAnnotations` set.
+   */
+  visibleAnnotations: number;
```

Comment:
I'm exposing this to later fix https://github.com/hypothesis/h/issues/9843.

I considered fixing that here, but it requires some work to make sure the fallback shows after the transition which hides the last card, so I'm delaying it to its own piece of work.